torch/_inductor/fx_passes/freezing_patterns.py

```diff
@@ -0,0 +1,131 @@
+import torch
+
+from .. import config
+from ..pattern_matcher import (
+    _return_true,
+    init_once_fakemode,
+    PatternMatcherPass,
+    register_graph_pattern,
+    stable_topological_sort,
+)
+
+# aten = torch.ops.aten
+
+# First pass_patterns[0] are applied, then [1], then [2]
+pass_patterns = [
+    PatternMatcherPass(),
+    PatternMatcherPass(),
+]
+
+
+def freezing_passes(gm: torch.fx.GraphModule):
+    """
+    Passes that run on after grad.  This is called once on the forwards
+    graph and once on the backwards graph.
+
+    The IR here has been normalized and functionalized.
+    """
+
+    if config.pattern_matcher:
+        lazy_init()
+
+    for patterns in pass_patterns:
+        patterns.apply(gm.graph)
+
+    if torch._C._has_mkldnn:
+        from .mkldnn_fusion import _eliminate_duplicate_packed_nodes
+
+        _eliminate_duplicate_packed_nodes(gm)
+
+    stable_topological_sort(gm.graph)
+    gm.recompile()
+    gm.graph.lint()
+
+
+@init_once_fakemode
+def lazy_init():
+    if torch._C._has_mkldnn:
+        from .mkldnn_fusion import _mkldnn_weight_pack_init
+
+        _mkldnn_weight_pack_init()
+
+    # addmm_patterns_init()
+
+
+def register_freezing_graph_pattern(
+    pattern, extra_check=_return_true, pass_number=1, prepend=False
+):
+    return register_graph_pattern(
+        pattern,
+        extra_check=extra_check,
+        pass_dict=pass_patterns[pass_number],
+        prepend=prepend,
+    )
+
+
+"""
+@functools.lru_cache(None)
+def addmm_patterns_init():
+    if torch.cuda.is_available():
```

Comment:
That code will be rebased after https://github.com/pytorch/pytorch/pull/100746 is merged.